### PR TITLE
[build] improve Xamarin.Android.sln usability in VS Windows

### PR DESCRIPTION
**Pull request by jonathanpeppers**
Xamarin.Android.sln has two issues that make `Xamarin.Android.sln` not
work too well in VS Windows:

* It builds MSBuild tasks and uses them within the same solution, such
  as `xa-prep-tasks.csproj` and
  `Xamarin.Android.Tools.BootstrapTasks.csproj`.
* We have conditional `@(ProjectReference)`

To improve things, I completely removed the `$(XAIntegratedTests)`
MSBuild property and any `@(ProjectReference)` that were conditional.

I moved the two bootstrap `.csproj` files to a new
`Xamarin.Android.BootstrapTasks.sln` which gets built during `make
prepare` or `msbuild Xamarin.Android.sln /t:Prepare`.

VS Windows was also complaining about the
`CreateManifestResourceNames` target missing for projects that merely
import `Microsoft.Common.targets` such as `r8.csproj`. I added an
empty target in `Configuration.props` to fix all cases of this.

Unfortunately, things are still not perfect as you can hit errors such
as:

    Unable to copy file "obj\Debug\Xamarin.Android.Build.Tasks.dll" to "C:\src\xamarin-android\bin\Debug\lib\xamarin.android\xbuild\Xamarin\Android\Xamarin.Android.Build.Tasks.dll".
    The process cannot access the file 'C:\src\xamarin-android\bin\Debug\lib\xamarin.android\xbuild\Xamarin\Android\Xamarin.Android.Build.Tasks.dll' because it is being used by another process.
    Could not copy "obj\Debug\Xamarin.Android.Build.Tasks.dll" to "C:\src\xamarin-android\bin\Debug\lib\xamarin.android\xbuild\Xamarin\Android\Xamarin.Android.Build.Tasks.dll".
    Exceeded retry count of 10. Failed. The file is locked by: "Microsoft Visual Studio 2019 (26660), MSBuild.exe (18800), MSBuild.exe (17416)"	Xamarin.Android.Build.Tasks

This occurs if you modify `Xamarin.Android.Build.Tasks.csproj`.

I am not yet sure if we can get around this remaining problem, but we
can revisit in the future.

### CODE REVIEW

**Comment by jonpryor** [comment]
Do we have an explanation for the file sharing issue on `Xamarin.Android.Build.Tasks.dll`?  Why is it being held open?

**Comment by jonathanpeppers** [comment]
VS Windows keeps MSBuild nodes alive, and if they have previously loaded `Xamarin.Android.Build.Tasks.dll` the assembly is locked.

At least under .NET framework, it seems like they are just using `Assembly.LoadFrom` in MSBuild: https://github.com/microsoft/msbuild/search?q=assembly.loadfrom

File: Makefile [hunk 1]
```diff
@@ -198,6 +202,7 @@ prepare-build: prepare-build-init
 .PHONY: prepare
 prepare: prepare-build
 	mono --debug $(PREPARE_EXE) $(_PREPARE_ARGS)
+	msbuild $(BOOTSTRAP_MSBUILD_FLAGS) $(BOOTSTRAP_SOLUTION)
```
Comment (jonpryor) [inline Makefile]: Why do we need this change on *Windows*?

File: Xamarin.Android.sln [hunk 2]
```diff
@@ -71,10 +69,6 @@ Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "sqlite-xamarin", "src\sqlit
 EndProject
 Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "OpenTK", "src\OpenTK-1.0\OpenTK.csproj", "{5EB9E888-E357-417E-9F39-DDEC195CE47F}"
 EndProject
-Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "create-bundle", "build-tools\create-bundle\create-bundle.csproj", "{1640725C-4DB8-4D8D-BC96-74E688A06EEF}"
```
Comment (jonpryor) [inline Xamarin.Android.sln]: I think we could also *remove* `build-tools/create-bundle`, though perhaps that should be a separate PR?  If it should be in a separate PR, I don't see why we'd need to remove `create-bundle` here.

File: Xamarin.Android.sln [hunk 3]
```diff
@@ -137,6 +131,7 @@ Global
 	GlobalSection(SharedMSBuildProjectFiles) = preSolution
 		src\Xamarin.Android.NamingCustomAttributes\Xamarin.Android.NamingCustomAttributes.projitems*{3f1f2f50-af1a-4a5a-bedb-193372f068d7}*SharedItemsImports = 4
 		src\Xamarin.Android.Build.Tasks\Tests\Xamarin.Android.Build.Tests\Xamarin.Android.Build.Tests.Shared.projitems*{53e4abf0-1085-45f9-b964-dcae4b819998}*SharedItemsImports = 4
+		src\Xamarin.Android.NamingCustomAttributes\Xamarin.Android.NamingCustomAttributes.projitems*{66cf299a-ce95-4131-bcd8-db66e30c4bf7}*SharedItemsImports = 4
```
Comment (jonpryor) [inline Xamarin.Android.sln]: I'm no expert on .sln syntax but...  isn't this line nearly duplicative of the following line, with just the GUID changed?  Do we really need both?

**Comment by jonpryor** [comment]
I think we need to do the "same" thing in Java.Interop -- not that I'm asking @jonathanpeppers to do so -- as that may help the VSMac experience.  (I think it'll also be needed to land PR #3393...)

File: Xamarin.Android.sln (hunk 3 above)
Comment (jonathanpeppers) [reply Xamarin.Android.sln]: I usually let VS Windows do what it wants with SLN files, but I removed this because I don't know what it is...

File: Makefile (hunk 1 above)
Comment (jonathanpeppers) [reply Makefile]: This PR removes the "bootstrap" MSBuild tasks from the SLN, so we need `make prepare` to build them now.

I think it makes sense to build them at that step? or do you think `make all` should do that?
